metrics/mobile-app-metrics.md

```diff
@@ -0,0 +1,108 @@
+
+For a list of terms used in this document, see [Glossary](./glossary.md).
+
+## Problem Description
+
+As Mobile Apps are developed, improved and iterated on, different versions of those Apps can be used on many different devices. System metrics data becomes invaluable for developers to observe the landscape of application installations and make sure further development and deployment of backend services will not affect users.
+
+## Proposed Solution
+
+A Metrics Service will be available from the Service Catalog, which can be provisioned by developers for use with their mobile applications.
+
+The extra components outlined in this proposal will be added to the existing components outlined in the [Metrics Endpoints and Auto Discovery Proposal](./mobile-app-metrics-service.md).
+
+The following diagram illustrates the components of the Metrics Service:
+
+![service diagram](./app-service-diagram.jpg)
+
+### Mobile App Client
+
+A native, platform-specific SDK for targetting the Metrics Service will be available to developers.
+
+Upon installation into a project and providing configuration for targetting the Metrics Service the SDK will automatically send general information about the general runtime for this application with a unique device ID, namely the `[SDK Version, clientID, timestamp]` tuple upon application initialization.
+
+The SDK will also instrument calls at the network level in order to gather metrics around communication and bandwith consumption.
+
+Other system resources utilization data can also be sent periodically.
+
+### Mobile Metrics Service
+
+A thin HTTP-based service will be present to receive metrics data from the mobile applications in an storage-agnostic way, then forwarding to the Storage service.
+
+Compared to mobile applications interfacing directly with the Storage service, this presents the following tradeoff:
+
+#### Pros
+
+- Possibility to upgrade or change the Storage service independently of mobile client SDK version and implementation;
+
+- Possible support for different Storage engines which can be selected based on application needs;
+
+#### Cons
+
+- Extra service deployment and scaling needs;
+
+- Development and maintenance of the new service;
+
+- Types of gathered metrics can become a subset of the underlying Storage's full range of types.
+
+#### HTTP Endpoints
+
+At a PoC level, the endpoints available from the Mobile Metrics Service should be:
+
+`POST /sdk-version`
+
+Receives `Content-Type: application/json` containing the SDK Version and a clientID that should be called from the SDK initialization process. A timestamp can be generated server-side.
+
+### Authentication for the Mobile Metrics Service
+
+The HTTP endpoints for the Mobile Metrics Service is going to be accessbile only via HTTPS to devices that provide an authentication header containing an appropriate API Key.
+
+This key will be acessible to the SDK via new fields in the generated `mobile-services.json` from the mobile CLI.
```

Comment:
@david-martin @wtrocki @pb82
Should we think about this right now? Is it part of the epic?

Reply:
There is no card on the trello board about securing the sdk metrics. @david-martin should we add a card for this?